### ISSUE TITLE
[a11y]: cds-textarea helper text not properly associated

### ISSUE THREAD

**Issue by shleewhite** [carbon-design-system/carbon#22363]
### Package

@carbon/web-components

### Package version

v2.55.0

### Automated testing tool and ruleset

Manual

### Description

Currently, the Carbon Web Component cds-textarea does not correctly associate the helper/error text and counter text with the input.

The aria-describedby attribute should be set to the ids of the helper/error text and the counter. In this screenshot from the storybook you can see that the attribute is missing

<img width="602" height="220" alt="Image" src="https://github.com/user-attachments/assets/a04f7db1-038f-47ec-a721-87470eb76cd8" />

For the counter, the content should not be within a label tag. It should be a div or span with an id. In this screenshot, you can see that the counter text is in a label not associated with the input.

<img width="557" height="240" alt="Image" src="https://github.com/user-attachments/assets/ebd94f56-64d9-4e2b-a611-81259aec84ff" />

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/docs/components-text-area--overview

### Steps to reproduce

1. Look at the public web component storybook page for textarea
2. Inspect the helper text, notice that there is no id on the element
3. Inspect the counter text, notice it is a label HTML element and it doesnt have an id
4. Inspect the textarea, notice that it is missing the aria-describedby attribute set to "{helper text id} {counter id}"

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
